dotm_search.py

```diff
@@ -1,15 +1,50 @@
 #!/usr/bin/env python
 # -*- coding: utf-8 -*-
+import os
+import argparse
+import zipfile
+parser = argparse.ArgumentParser()
+
 
 """
 Given a directory path, search all files in the path for a given text string
 within the 'word/document.xml' section of a MSWord .dotm file.
 """
-__author__ = "???"
+__author__ = "knmarvel"
+#   time spent on this project: 2 hrs
+
+
+def examine_files(dir, char):
+    files = {}
+    total_files_searched = 0
+    for filename in os.listdir(dir):
+        filepath = dir + "/" + filename
+        if zipfile.is_zipfile(filepath):
+            with zipfile.ZipFile(filepath, 'r') as zip_ref:
+                zip_ref.extractall("./extracted")
+            with open('./extracted/word/document.xml', "r") as doc:
+                doc = doc.read()
+            if "office/word" in doc:
+                total_files_searched += 1
+                if char in doc:
+                    files[filename] = doc[doc.index(char) - 40: doc.index(char) + 40]
```

Comment:
what happens if char is at index 10?  or is only 10 characters before the end of the file?

Reply:
A MS Word file has headers and footers that are at least 500 characters. Never will the string be less than 80 characters and be in a MS word file.